server/test/views/applicant/EditOrDiscardAnswersModalCreatorTest.java

```diff
@@ -0,0 +1,93 @@
+package views.applicant;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static play.test.Helpers.fakeApplication;
+import static play.test.Helpers.fakeRequest;
+import static views.questiontypes.ApplicantQuestionRendererParams.ErrorDisplayMode.DISPLAY_ERRORS;
+import static views.questiontypes.ApplicantQuestionRendererParams.ErrorDisplayMode.DISPLAY_ERRORS_WITH_MODAL_PREVIOUS;
+import static views.questiontypes.ApplicantQuestionRendererParams.ErrorDisplayMode.DISPLAY_ERRORS_WITH_MODAL_REVIEW;
+import static views.questiontypes.ApplicantQuestionRendererParams.ErrorDisplayMode.HIDE_ERRORS;
+
+import auth.CiviFormProfile;
+import auth.CiviFormProfileData;
+import controllers.applicant.ApplicantRoutes;
+import java.util.Set;
+import org.junit.Test;
+import play.i18n.Lang;
+import play.i18n.MessagesApi;
+import play.libs.concurrent.HttpExecutionContext;
+import repository.AccountRepository;
+import repository.DatabaseExecutionContext;
+import repository.ResetPostgres;
+import services.applicant.ApplicantPersonalInfo;
+import services.applicant.Block;
+import services.cloud.ApplicantStorageClient;
+import services.settings.SettingsManifest;
+import views.ApplicationBaseView;
+import views.questiontypes.ApplicantQuestionRendererParams;
+
+public class EditOrDiscardAnswersModalCreatorTest extends ResetPostgres {
+  private final EditOrDiscardAnswersModalCreator modalCreator =
+      new EditOrDiscardAnswersModalCreator();
+
+  @Test
+  public void createModal_modeDisplayErrors_throws() {
+    ApplicationBaseView.Params params = makeParamsWithMode(DISPLAY_ERRORS);
+    assertThatThrownBy(() -> modalCreator.createModal(params))
+        .isInstanceOf(IllegalArgumentException.class);
+  }
+
+  @Test
+  public void createModal_modeHideErrors_throws() {
+    ApplicationBaseView.Params params = makeParamsWithMode(HIDE_ERRORS);
+    assertThatThrownBy(() -> modalCreator.createModal(params))
+        .isInstanceOf(IllegalArgumentException.class);
+  }
+
+  @Test
+  public void createModal_modeDisplayErrorsWithModalReview_ok() {
+    ApplicationBaseView.Params params = makeParamsWithMode(DISPLAY_ERRORS_WITH_MODAL_REVIEW);
+    modalCreator.createModal(params);
+    // No assert, just need no crash
+  }
+
+  @Test
+  public void createModal_modeDisplayErrorsWithModalPrevious_ok() {
+    ApplicationBaseView.Params params = makeParamsWithMode(DISPLAY_ERRORS_WITH_MODAL_PREVIOUS);
+    modalCreator.createModal(params);
+    // No assert, just need no crash
```

Comment:
**thought**: junit 5 has `assertDoesNotThrow()` but I think we're on 4, unfortunately

Reply:
Sad :(